[BUG] 4 mods together causes invalid main product light-oil. Angels bio processing involved

**Describe the bug**
Enabling Omniwood, Bobs Greenhouses, Yuoki, and Angels bio processing with their required base mods causes the following error.
```
Error ModManager.cpp:1508: Error while loading recipe prototype "y_ft_recipe" (recipe): Difficulty normal: Recipe has invalid main product light-oil. The product does not exist in the recipe product list.
Modifications: Yuoki Industries › Omniwood › Angel's Petro Chemical Processing
```
Remove any one mod from that group of 4 and the game loads properly.

Information to reproduce the behavior:
1. Game Version 18.30
2. Modlist
Omniwood, Bobs Greenhouses, Yuoki, and Angels bio processing with their required mods.

--- a/angelsrefining/changelog.txt
+++ b/angelsrefining/changelog.txt
@@ -3,6 +3,7 @@ Version: 0.11.14
 Date: ???
   Bugfixes:
     - Fixed marathon mode text (289)
+    - Replacing main_product didn't work well (286)
 ---------------------------------------------------------------------------------------------------
 Version: 0.11.13
 Date: 05.06.2020

--- a/angelsrefining/prototypes/override-functions.lua
+++ b/angelsrefining/prototypes/override-functions.lua
@@ -675,13 +675,13 @@ local function adjust_recipe(recipe, k) -- check a recipe for basic adjustments 
         end
         if replace[item.name] then
           if item.probability then
-             if replace[item.name].probability and replace[item.name].probability ~= item.probability then
-              replace[item.name].probability = item.probability
+            if replace[item.name].probability and replace[item.name].probability ~= item.probability then
               --update probability if it exists in both cases
-             else
-              replace[item.name .."-p"]=item
+              replace[item.name].probability = item.probability
+            else
               --skip, don't touch recipes that add a probability to a static
-             end
+              replace[item.name .. "-p"] = item
+            end
           elseif item.amount ~= replace[item.name].amount then --check both have amount and update old to new
             replace[item.name].amount = item.amount
           end
@@ -699,7 +699,7 @@ local function adjust_recipe(recipe, k) -- check a recipe for basic adjustments 
     adjust_subtable(path, "ingredients", "recipe_items")
     adjust_member(path, "result", "recipe_items")
     adjust_subtable(path, "results", "recipe_items")
-    adjust_member(recipe, "main_product", "recipe_items")
+    adjust_member(path, "main_product", "recipe_items")
   end
   if recipe.category ~= "angels-converter" then -- leave converter recipes alone so we can still use them if necessary
     if recipe.normal or recipe.expensive then

